### PR TITLE
Propagate sampled flag

### LINKED ISSUE #261: Sampling decision propagation
It doesn't appear that the library supports sampling decision propagation. It doesn't take the sampled flag from the carrier, nor propagate it locally nor used it in collection.

### PR DESCRIPTION
**Pull request by ocelotl**
Fixes #261

### CODE REVIEW

**Comment by kayousterhout** [comment]
Do you need to set Sampled here (instead of always using true): https://github.com/lightstep/lightstep-tracer-go/blob/master/propagation_lightstep.go#L40

**Comment by ocelotl** [comment]
> Do you need to set Sampled here (instead of always using true): https://github.com/lightstep/lightstep-tracer-go/blob/master/propagation_lightstep.go#L40

Right, fixed. :+1:

File: tracer_test.go [hunk 1]
```diff
@@ -348,6 +348,31 @@ var _ = Describe("Tracer", func() {
...
+				span.Finish()
+				Expect(fakeRecorder.RecordSpanCallCount()).To(Equal(0))
+			})
+		})
+
+		Context("when the parent span carries the Sampled flag", func() {
+			It("should propagate flags from parent to child", func() {
+				opentracing.SetGlobalTracer(tracer)
+
+				opentracing_parent_span := tracer.StartSpan("parent", SetTraceID(1), SetSampled("false"))
```
Comment (codeboten) [inline tracer_test.go:366]: Please add a test here to ensure that if a span is started with the sampled set to anything else than false, the child span will be recorded.

**Review by codeboten — CHANGES_REQUESTED** [review]
Change looks good, just one comment around one additional test that I think we should have.

File: tracer_test.go (hunk 1 above)
Comment (ocelotl) [reply tracer_test.go:378]: Fixed

**Review by codeboten — APPROVED** [review]
👍

**Pull request merged**